File: backend/internal/domain/entity/message.go
package entity

import (
	"errors"
	"time"
)

// MessageType represents the type of message
type MessageType string

const (
	MessageTypeText  MessageType = "message"
	MessageTypeError MessageType = "error"
)

// Message represents a chat message entity
type Message struct {
	Type      MessageType `json:"type"`
	Content   string      `json:"content"`
	Sender    string      `json:"sender"`
	Timestamp time.Time   `json:"timestamp"`
}

// NewMessage creates a new message with validation
func NewMessage(msgType MessageType, content, sender string) (*Message, error) {
	if content == "" {
		return nil, errors.New("message content cannot be empty")
	}
	if sender == "" {
		return nil, errors.New("message sender cannot be empty")
	}

	return &Message{
		Type:      msgType,
		Content:   content,
		Sender:    sender,
		Timestamp: time.Now(),
	}, nil
}

// Validate validates the message
func (m *Message) Validate() error {
	if m.Content == "" {
		return errors.New("message content cannot be empty")
	}
	if m.Sender == "" {
		return errors.New("message sender cannot be empty")
	}
	return nil
}

File: frontend/src/domain/entities/Message.js
// Message entity
export class Message {
    constructor(type, content, sender, timestamp = new Date()) {
        this.type = type;
        this.content = content;
        this.sender = sender;
        this.timestamp = timestamp;
    }

    static fromJSON(json) {
        return new Message(
            json.type,
            json.content,
            json.sender,
            json.timestamp ? new Date(json.timestamp) : new Date()
        );
    }

    toJSON() {
        return {
            type: this.type,
            content: this.content,
            sender: this.sender,
            timestamp: this.timestamp.toISOString()
        };
    }

    validate() {
        if (!this.content || this.content.trim() === '') {
            throw new Error('Message content cannot be empty');
        }
        if (!this.sender) {
            throw new Error('Message sender cannot be empty');
        }
        return true;
    }
}

export const MessageType = {
    TEXT: 'message',
    ERROR: 'error'
};

export const MessageSender = {
    USER: 'user',
    AI: 'ai',
    SYSTEM: 'system'
};
